q3 survey

pq opened the pull request:
/cc @jayoung-lee @DanTup to double-check URLs, etc.

@devoncarew

devoncarew commented on src/io/flutter/survey/FlutterSurveyNotifications.java:
```diff
@@ -0,0 +1,162 @@
+  private static final String FLUTTER_LAST_SURVEY_CHECK_KEY = "FLUTTER_LAST_SURVEY_CHECK_KEY";
+  private static final long CHECK_INTERVAL_IN_MS = TimeUnit.HOURS.toMillis(40);
+
+  private static final String SURVEY_ACTION_TEXT = "Take Flutter Q3 Survey";
```
`"Take Flutter Q3 Survey"` ==> `"Take Flutter's Q3 Survey"`? Not sure where or if this string is ever displayed to users

pq replied:
I'm great with these text improvements but should defer to @jayoung-lee.  This would also diverge from VS Code...

jayoung-lee replied:
Does this get displayed to the users? I thought we are using "Help improve Flutter! Take our Q3 survey." for the pop up message.

pq replied:
@jayoung-lee: you're correct.  The visible text is "Help improve Flutter! Take our Q3 survey."   (The other text is an internal label for the action.)

pq replied:
EDIT: it does.  Trimmed to "Take survey"

![image](https://user-images.githubusercontent.com/67586/61814844-d269f000-adfd-11e9-9a4a-7c52632a7634.png)

devoncarew commented on src/io/flutter/survey/FlutterSurveyNotifications.java:
```diff
@@ -0,0 +1,162 @@
+  private static final String SURVEY_TAKEN = "io.flutter.survey.2019.q3.alreadyTaken";
+  private static final String SURVEY_URL = "https://google.qualtrics.com/jfe/form/SV_3kiGXYfYOfXUjB3?Source=IntelliJ&ClientID=";
+
+  private static long SURVEY_START;
```
Perhaps `SURVEY_START_MS_EPOCH` or similar?

pq replied:
👍

devoncarew commented on src/io/flutter/survey/FlutterSurveyNotifications.java:
```diff
@@ -0,0 +1,162 @@
+    final long lastCheckedMillis = properties.getOrInitLong(FLUTTER_LAST_SURVEY_CHECK_KEY, 0);
+    if (System.currentTimeMillis() - lastCheckedMillis < CHECK_INTERVAL_IN_MS) return;
+
+    properties.setValue(FLUTTER_LAST_SURVEY_CHECK_KEY, String.valueOf(System.currentTimeMillis()));
```
I assume that `properties` values can only be strings (in terms of why you're not storing as a long)?

pq replied:
👍

devoncarew commented on src/io/flutter/survey/FlutterSurveyNotifications.java:
```diff
@@ -0,0 +1,162 @@
+      null
+    );
+
+    notification.addAction(new AnAction(SURVEY_ACTION_TEXT) {
```
Ah, here's where this string is displayed. Perhaps something like `Sure` or `Sure!` (or just something more plain but terser than the current text, like `Take survey`)? That would go well with the `No thanks` below.

pq replied:
@jayoung-lee ?

jayoung-lee replied:
Either way is fine :)

pq replied:
Trimmed.  Thanks!

DanTup commented:
On the surface it all looks reasonable to me, but there's so much I don't know about Java and this project that I'm not sure that's all that useful :-)

devoncarew commented on src/io/flutter/survey/FlutterSurveyNotifications.java:
```diff
@@ -0,0 +1,162 @@
+  private static final String SURVEY_TITLE = "Help improve Flutter! Take our Q3 survey.";
+
+  private static final String SURVEY_TAKEN = "io.flutter.survey.2019.q3.alreadyTaken";
+  private static final String SURVEY_URL = "https://google.qualtrics.com/jfe/form/SV_3kiGXYfYOfXUjB3?Source=IntelliJ&ClientID=";
```
You may want to remove the `&ClientID=` suffix, and only add it if also appending a client ID.

pq replied:
This was the format @jayoung-lee suggested.  Not sure what's most desirable for easy processing...

jayoung-lee replied:
Either way (&ClientID="" or nothing) will give me the same result to me. So please do whatever works better for you.

pq replied:
Sounds good.  Thanks!

devoncarew commented:
> This was the format @jayoung-lee suggested. Not sure what's most desirable for easy processing...

Gotcha, but this:

> https://google.qualtrics.com/jfe/form/SV_3kiGXYfYOfXUjB3?Source=IntelliJ&ClientID=

has a hanging GET variable.

jayoung-lee commented:
Please see my comments above! Thanks for getting me involved.

pq commented:
Thanks for all the feedback!

I updated the dialog text and removed the dangling GET.

![image](https://user-images.githubusercontent.com/67586/61813066-b8c6a980-adf9-11e9-94fa-3f14ad0346a3.png)

👍

jayoung-lee commented:
> Thanks for all the feedback!
> 
> I updated the dialog text and removed the dangling GET.
> 
> ![image](https://user-images.githubusercontent.com/67586/61813066-b8c6a980-adf9-11e9-94fa-3f14ad0346a3.png)
> 
> 👍

This will show “By clicking on this link you agree to share feature usage along with the survey responses.” when there's clientID to append, right?

pq commented:
Yep.  Thanks for checking @jayoung-lee !

![image](https://user-images.githubusercontent.com/67586/61814490-0ee91c00-adfd-11e9-8e0a-d2ab3b560a63.png)

DanTup commented:
I've tweaked the action text in VS Code to just "Take Survey" to match 👍

pq commented:
Awesome.  Thanks @DanTup!
